budgetportal/tests/test_prov_infra_projects.py

```diff
@@ -252,14 +533,175 @@ def test_url_path(self):
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertContains(response, name)
 
+    def test_facet_url_path(self):
+        province = "Eastern Cape"
+        data = {"selected_facets": "province_exact:{0}".format(province)}
+        response = self.client.get(self.facet_url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        result = response.data["objects"]["results"][0]
+        url_path = result["url_path"]
+        response = self.client.get(url_path)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, province)
 
-class ProvInfraProjectContentTestCase(APITestCase):
-    fixtures = ["test-prov-infra-project-content"]
 
-    def test_project_detail_content(self):
-        project = ProvInfraProject.objects.first()
-        url = project.get_absolute_url()
+class ProvInfraProjectSnapshotTestCase(APITransactionTestCase):
+    def setUp(self):
+        self.project = ProvInfraProject.objects.create(IRM_project_id=1)
+        self.fin_year = FinancialYear.objects.create(slug="2030-31")
+        self.quarter_1 = Quarter.objects.create(number=1)
+        self.irm_snapshot_1 = IRMSnapshot.objects.create(
+            financial_year=self.fin_year,
+            quarter=self.quarter_1,
+            date_taken=date(year=2019, month=1, day=1),
+        )
+        self.project_snapshot_1 = ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=self.irm_snapshot_1,
+            project=self.project,
+            status="Construction",
+        )
+
+        self.quarter_2 = Quarter.objects.create(number=2)
+        self.irm_snapshot_2 = IRMSnapshot.objects.create(
+            financial_year=self.fin_year,
+            quarter=self.quarter_2,
+            date_taken=date(year=2019, month=6, day=1),
```

Comment:
`date_taken` is irrelevant in these tests - please make them identical to demonstrate that.

Reply:
Solved.

https://github.com/vulekamali/datamanager/pull/213/commits/ce33ff33180f1d00ef558beb0e0ecbeb1f429674